src/components/builder/comparison-value-input.tsx

```diff
@@ -0,0 +1,60 @@
+import {useFormikContext} from 'formik';
+import {ExtendedComponentSchema, Utils as FormioUtils} from 'formiojs';
+import {useContext, useEffect, useState} from 'react';
+import usePrevious from 'react-use/esm/usePrevious';
+
+import {BuilderContext} from '@/context';
+import {getRegistryEntry} from '@/registry';
+import Fallback from '@/registry/fallback';
+import {comparisonValueComponentProps} from '@/registry/types';
+
+export interface simpleConditional {
+  conditional: {
+    eq: string;
+    when: string;
+    show: null | boolean;
+  };
+}
+
+export const ComparisonValueInput: React.FC = () => {
+  const {getFormComponents} = useContext(BuilderContext);
+  const {values, setFieldValue} = useFormikContext<simpleConditional>();
+
+  const [chosenComponent, setChosenComponent] = useState<ExtendedComponentSchema | null>(null);
+  const previousWhen = usePrevious(values.conditional.when);
+
+  useEffect(() => {
+    const conditional = values.conditional;
+    if (conditional?.when === '' && conditional?.eq !== '') {
+      setFieldValue('conditional.eq', '');
+      setChosenComponent(null);
+      return;
+    }
+
+    if (previousWhen !== conditional?.when) {
+      setFieldValue('conditional.eq', '');
+    }
+
+    const component: ExtendedComponentSchema = FormioUtils.getComponent(
+      getFormComponents(),
+      conditional.when,
+      false
+    );
+
+    setChosenComponent(component);
+  }, [values, setFieldValue]);
+
+  if (!chosenComponent) return null;
```

Comment:
Probably this doesn't create issues in our case, but it seems to be better [to return `false` instead](https://medium.com/@davidkelley87/stop-using-return-null-in-react-a2ebf08fc9cd)

Reply:
After further discussion, we decided to stick to `null` since the react docs say to use null: https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null

Reply:
I don't know who to trust anymore in this frontend world :shrug: :shrug: :shrug: :shrug:

Reply:
The `React.FC` type declarations are a good starting point :P they don't allow `false` as return value

Reply:
Ok well reading the comments of the article seems to be a good starting point as well